packages/dds/merge-tree/src/attributionCollection.ts

```diff
@@ -85,6 +85,18 @@ export interface IAttributionCollection<T> {
 	 */
 	getAtOffset(offset: number, channel?: string): AttributionKey | undefined;
 
+	/**
+	 * Retrieves all the [offset, attribution key] pairs for the provided offset range.
```

Comment:
this doc comment requires more details on semantics of the representation, perhaps with examples.

E.g. with the current api, it should be called out that:
- the returned array is sorted by offset
- it may include offsets outside the start/end offset arguments, in cases where attributionkey boundaries don't align exactly with startOffset/endOffset (provide an example, e.g. the one from your test case)

Also, is there a reason we went with a tuple here rather than e.g. `{ offset: number; key: AttributionKey }[]`?

Reply:
Went for the 2d array instead of array of objects just to prevent duplication of info on objects of same kind. Like the user knows when using the API by comment that first will be offset and second will be attribution key. 
I was in double mind to choose any one and don't have a strong preference. I can replace with array of objects is you would like and see benefit.

Reply:
But lookwise this definitely looks better: { offset: number; key: AttributionKey }[]

Reply:
Also, in response, the first offset could be lower than the startOffset but no offset would be greater than the endOffset.

Reply:
Changed the return type.